parsl/tests/test_providers/test_slurm_template.py

```diff
@@ -0,0 +1,29 @@
+import logging
+import random
+
+from unittest import mock
+import pytest
+
+from parsl.channels import LocalChannel
+from parsl.providers import SlurmProvider
+
+
+@pytest.mark.local
+def test_submit_script_basic(tmp_path):
+    """Test slurm resources table"""
+
+    provider = SlurmProvider(
+        partition="debug", channel=LocalChannel(script_dir=tmp_path)
+    )
+    provider.script_dir = tmp_path
+    job_id = str(random.randint(55000, 59000))
+    provider.execute_wait = mock.MagicMock()
+    provider.execute_wait.return_value = (0, f"Submitted batch job {job_id}", "")
```

Comment:
Where feasible, I suggest that we attempt to get in the habit of using `spec=` in the tests.  In this case, we know the precise `execute_wait` definition we expect, so perhaps:
```python
provider.execute_wait = mock.Mock(spec=SlurmProvider.execute_wait)
```
In the best case scenario, no change.  In the other best case scenario, we catch an unintended call to that method.

Reply:
Nice, I've updated the PR with this.